### ISSUE TITLE
Enable Datapusher on Staging

### ISSUE THREAD

**Issue by mbellotti** [OCHA-DAP/hdx-infrastructure#67]
It will need to be configured to work around localhost like our old friend dataproxy :)

**Comment by teodorescuserban** [comment]
Will do it today.

**Comment by teodorescuserban** [comment]
Well, not too much luck on that.
I created a nice little virtualenv for datapusher, read the docs, grab it, configure it. It appears it is working.
You can test it by going to stag internal ip address through vpn on port 8800), you should get
_{
  "help": "\n        Get help at:\n        http://ckan-service-provider.readthedocs.org/."
}_
However, it cant really push to ckan datastore because of the following error:
**IndexError: pop from empty list**
This led me to commenting out on https://github.com/ckan/datapusher/issues/17

Don't know what else to say/do :)
Please advise.

**Comment by teodorescuserban** [comment]
![datapusher-error](https://cloud.githubusercontent.com/assets/1542984/4203459/d6c973a8-382f-11e4-8a89-75c4ec92ada1.png)
Actually, it looks like my issue reporting is not so good. The datastore worked fine 8 days ago as well - no gremlins, sorry, just a beginning of Alzheimers :)
The error reported in issue is referring to a new resource.

Steps to reproduce:
1. Login as sysadmin in stag ckan.
2. Go to Airports in Zimbabwe dataset.
3. Download the airports.csv (the initial one - see its description)
4. Edit the above dataset and choose to add a resource
5. Add the previously downloaded airports.csv - datapusher should have push this to datastore
6. See the error (by going to newly added resource edit)

**Comment by teodorescuserban** [comment]
Update: it seems that this datapusher bug makes impossible for later datastore manual upload.
:|
So I cannot manually force the resource into datastore if that resource has been added while datapusher was enabled on ckan.
The manual upload worked for the first initial resource because that is either NOT in filestore at all or it comes from a prod db snapshot (on prod datapusher did not tried to upload it to datastore).

**Comment by rufuspollock** [comment]
@teodorescuserban what do you mean here? When you say a manual upload what do you mean? you mean pushing to the datastore API or something else ...?

**Comment by teodorescuserban** [comment]
I mean clicking the big blue buton that readus "Upload to datastore". :)

**Comment by teodorescuserban** [comment]
Update: trying to add a new url type resource works!

So this narrows down the issue a bit. This issue appears only when trying to add a new resource by uploading a file.

**Comment by teodorescuserban** [comment]
Update:

I had a feeling it has to do with our http-to-https redirect on our localstore (maybe datapusher would not handle well the 301 and 302 redirects) - it is not the case.

The issue is that our web server serving the files from the localdatastore sent them with a **text/html** content type header. For some reason datapusher bailed out on that. 

I checked what mime type I get from the first resource in the dataset (the one with the url to ourairport.com) and got the mime type **text/csv**.

I have added this mime type to the mime type of our web server and now it works flawlessly.

Assigning to @cjhendrix to close it at his own discretion (maybe you feel more testing is required?)
